### PR TITLE
gcs: fix integer overflow on 32-bit systems

### PR DESCRIPTION
**Pull request by aakselrod**

### CODE REVIEW

File: gcs/gcs.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ func BuildGCSFilter(P uint8, key [KeySize]byte, data [][]byte) (*Filter, error)
 	if len(data) == 0 {
 		return nil, ErrNoData
 	}
-	if len(data) > ((1 << 32) - 1) {
+	if len(data) > int(^uint32(0)) {
```
Comment (halseth) [inline gcs/gcs.go:63]: This should maybe have a comment?

**Review by Roasbeef — COMMENTED** [review]
Diff looks good to me, still need to verify manually that this fixes the compilation issue on 32-bit systems. 

I agree with @halseth that the new diff could use a brief commenting explaining the rationale behind the syntax used.

**Comment by halseth** [comment]
Still does not work for me:
```
# github.com/lightningnetwork/lnd/vendor/github.com/roasbeef/btcutil/gcs
vendor/github.com/roasbeef/btcutil/gcs/gcs.go:63: constant 4294967295 overflows int
```

**Comment by KanoczTomas** [comment]
Hello, 

@halseth helped me figure out my build on a raspberry pi.

editing line 63 as such: 
 if len(data) > math.MaxInt32 {

helped me. I had to include the math package at the beginning, tried to put in 1<<31 - 1 instead of math.MaxInt32 (as it is a defined as such in the math pkg) and it fixed the build issue for me.

So both approaches work ... if len(data) > (1<<31 - 1) { is probably easier as fix at the moment, as there is no need add the math package as well

**Comment by Roasbeef** [comment]
Pulled in that fixed locally, pushing shortly. 

Closing.

**Pull request closed without merging**